core/src/main/java/com/onelogin/saml2/authn/SamlResponse.java

```diff
@@ -297,10 +297,10 @@ public boolean isValid(String requestId) {
 				}
 
 				// Check the session Expiration
-				DateTime sessionExpiration = this.getSessionNotOnOrAfter();
+				Instant sessionExpiration = this.getSessionNotOnOrAfter();
 				if (sessionExpiration != null) {
-					sessionExpiration = sessionExpiration.plus(Constants.ALOWED_CLOCK_DRIFT * 1000);
-					if (sessionExpiration.isEqualNow() || sessionExpiration.isBeforeNow()) {
+					sessionExpiration = ChronoUnit.SECONDS.addTo(sessionExpiration, Constants.ALOWED_CLOCK_DRIFT);
```

Comment:
Seem like this pattern repeat a lot. Think it would be good to make `isOnOrAfterNow(Instant)` and `isBeforeNow(Instant)` methods to the util class. The methods would account for the allowed clock drift. 

It would make the code more readable.